src/main/java/picard/analysis/CollectQualityYieldMetrics.java

```diff
@@ -50,6 +50,8 @@
         programGroup = Metrics.class
 )
 public class CollectQualityYieldMetrics extends SinglePassSamProgram {
+    private QualityYieldMetricsCollector collector = new QualityYieldMetricsCollector();
```

Comment:
This is confusing...why not set it to null here, keep the argument defaults as they were and in setup call the right constructor?